cpu/atxmega/periph/gpio.c

```diff
@@ -0,0 +1,543 @@
+    uint8_t irq_state;
+    uint8_t tmp;
+
+    if (mode & GPIO_OUT) {
+        port->DIRSET = pin_mask;
+    }
+    else {
+        port->DIRCLR = pin_mask;
+    }
+
+    irq_state = irq_disable();
+
+    for (uint8_t p = 0; p < 8; p++) {
+        if (pin_mask & (1 << p)) {
+            tmp = (mode & ~PORT_ISC_gm);
+            tmp |= (mode & GPIO_ANALOG)
+                   ?  PORT_ISC_INPUT_DISABLE_gc
+                   :  ((flank >> 4) & PORT_ISC_gm);
+            pin_ctrl[p] = tmp;
+        }
+    }
+
+    if (flank & GPIO_INT_DISABLED_ALL) {
+        config_ctx[port_num + PORT_MAX].cb = NULL;
+        config_ctx[port_num + PORT_MAX].arg = NULL;
+        config_irq[port_num + PORT_MAX] = 0;
+        config_ctx[port_num].cb = NULL;
+        config_ctx[port_num].arg = NULL;
+        config_irq[port_num] = 0;
+
+        port->INTCTRL = 0;
+        port->INT1MASK = 0;
+        port->INT0MASK = 0;
+        port->INTFLAGS = PORT_INT1IF_bm | PORT_INT0IF_bm;
+    }
+    else if (flank & GPIO_INT1_VCT) {
+        config_ctx[port_num + PORT_MAX].cb = cb;
+        config_ctx[port_num + PORT_MAX].arg = arg;
+        config_irq[port_num + PORT_MAX] = (flank & GPIO_LVL_HIGH)
+                                          << PORT_INT1LVL_gp;
+
+        if ((flank & GPIO_LVL_HIGH) == GPIO_LVL_OFF) {
+            port->INT1MASK &= ~pin_mask;
+        }
+        else {
+            port->INT1MASK |= pin_mask;
+        }
+
+        port->INTFLAGS = PORT_INT1IF_bm;
+
+        /* Get mask from INT 0 and apply new INT 1 mask */
+        port->INTCTRL = (port->INTCTRL & PORT_INT0LVL_gm)
+                        | config_irq[port_num + PORT_MAX];
+    }
+    else {
+        config_ctx[port_num].cb = cb;
+        config_ctx[port_num].arg = arg;
+        config_irq[port_num] = (flank & GPIO_LVL_HIGH)
+                               << PORT_INT0LVL_gp;
+
+        if ((flank & GPIO_LVL_HIGH) == GPIO_LVL_OFF) {
+            port->INT0MASK &= ~pin_mask;
+        }
+        else {
+            port->INT0MASK |= pin_mask;
+        }
+
+        port->INTFLAGS = PORT_INT0IF_bm;
+
+        /* Get mask from INT 1 and apply new INT 0 mask */
+        port->INTCTRL = (port->INTCTRL & PORT_INT1LVL_gm)
+                        | config_irq[port_num];
+    }
+
+    irq_restore(irq_state);
+
+}
+
+int gpio_init(gpio_t pin, gpio_mode_t mode)
+{
+    DEBUG("gpio_init pin = 0x%02x mode = 0x%02x\n", pin, mode);
+
+    _gpio_pinctrl_set(pin, mode, GPIO_INT_DISABLED_ALL, NULL, NULL);
+
+    return 0;
+}
+
+int gpio_init_int(gpio_t pin, gpio_mode_t mode, gpio_flank_t flank,
+                  gpio_cb_t cb, void *arg)
+{
+    DEBUG("gpio_init_int pin = 0x%02x mode = 0x%02x flank = 0x%02x\n", pin,
+          mode, flank);
+
+    if (mode & GPIO_ANALOG) {
+        DEBUG("Pin can't be ANALOG (input buffer disabled) and INT at same time.\n");
+        return -1;
+    }
+
+    _gpio_pinctrl_set(pin, mode, flank, cb, arg);
+
+#if ENABLE_DEBUG
+    _print_config(pin);
+#endif
```

Comment:
```suggestion
    if (IS_ACTIVE(ENABLE_DEBUG)) {
        _print_config(pin);
    }
```

Note: With this I think you can drop `__attribute__((used))` in the declaration of the function above with this, as the function will now only be optimized out and not remain unreferenced.